### DISCUSSION TITLE
How to set Schema to No JSON Schema

### DISCUSSION THREAD

**Discussion by bitstarr** [redhat-developer/vscode-yaml#768] [Q&A] [answered]
When I open a site config for grav cms (like [this](https://github.com/bitstarr/grav-ddev-kickstart/blob/main/user/config/site.yaml)) VS Code will use 'Ansible Playbook' as schema, but I don't want to use any schema on that file.

Who do I disable this behaviuor? I'm so annoyed by this.

**Comment by gorkem** [comment, marked as answer]
This is because[ schema store catalog](https://www.schemastore.org/api/json/catalog.json) maps `site.yaml` to Ansible Playbook. One way to fix this is to set `yaml.schemaStore.enable` and disable automatic schema store matching. Alternately you can set the mapping for `site.yaml` explicitly on settings to override the schema store.

**Reply by gssor5** [reply]
###

**Comment by filcuk** [comment]
Just for anyone else looking for how to set the custom mapping, add the following to VSC `settings.json`:
```
"yaml.schemas": {
  "none": "site.yaml"
}
```

**Reply by dbowling** [reply]
I tried that suggestion, but it causes a different error in my linter: `Unable to load schema from '/none': No content.` 

Out of frustration, I wrote a schema that is wide open.

https://github.com/dbowling/null-schema

Now I apply this with:

```json
"yaml.schemas": {    
  "https://raw.githubusercontent.com/dbowling/null-schema/refs/heads/main/null.schema.json": [
    "**/kyverno/**/*"   
  ]
}
```

Hope it helps others.

**Comment by davetapley** [comment]
See also: 
- https://github.com/redhat-developer/vscode-yaml/issues/401#issuecomment-1398783128

**Comment by AndiDog** [comment]
Can we add a menu entry to choose "No schema"? Otherwise the whole editor is full of red underlines, making the file unreadable. I don't want to turn some global setting off since the feature is generally helpful. But if the auto-matching fails, I want to set "No schema" once.

**Reply by thardy** [reply]
That would be far too intelligent.  The ability to actually set the schema to "None" after clicking "Select JSON schema" in the IDE?  

You sir, are talking crazy.  It makes much more sense to force people to google changing a global setting or pollute their file with some functional comment than allow people to set the schema to "None"/"No Schema"/"No JSON schema" where they can choose the schema.  We have UX professionals for this, and they say you are wrong.  Users prefer to be mystified and have to google how to accomplish something rather than being able to figure out how to do something just by looking at it.

Away with you and your crazy ideas.

**Comment by lexicalunit** [comment]
I ran into this bug today because a file was named `rules.yml` which was auto-mis-detected as Prometheus configuration file. Certainly useful for that auto detection to exist. But it is also **exceedingly strange** that you can't choose "No JSON Schema" as an option in the `Select JSON Schema` menu and instead have to hack in a workaround. Seems like it would be an easy fix.

**Comment by legoGoat** [comment]
I also came across this bug today, and the shenanigans of setting it to "none" via either method has VSC complaining that it is `Unable to load schema from '/none': No content.`, which, _of course_.
I'm another vote for being able to choose a `No JSON Schema` option - that is clearly an option that exists, that's the one my other files have. Apparently assigning a JSON schema is turned on by default, so it would make sense to be able to undo it when VSC wrongly attaches one out of a sudden.

**Reply by dbowling** [reply]
@legoGoat Hope this helps: https://github.com/redhat-developer/vscode-yaml/discussions/768#discussioncomment-12568702

**Comment by JarlaxleXenocide** [comment]
Can we please add "No JSON Schema" when clicking on the Schema selection in the IDE? Yes, I know I can modify my settings, but this is not what this discussion is about. 

It was stated most directly by `thardy`:
https://github.com/redhat-developer/vscode-yaml/discussions/768#discussioncomment-7471478
